biztown/src/components/FormInputLayout.js

```diff
@@ -1,260 +1,428 @@
+        return { aboutData };
+      });
     }
+  };
 
-    handleAboutSectionChange = event => {
-        const {name, value} = event.target
-        if (name === "schoolDistrict") {
-            this.setInitialSchoolInfo(value);
-        } else {
-            this.setState(prevState => {
-                var aboutData = JSON.parse(JSON.stringify(prevState.aboutData));
-                aboutData[name] = value;
-                return { aboutData };
-            });
-        }
+  handleMultipleSectionChange = (event) => {
+    const { name, value } = event.target;
+    this.setState((prevState) => {
+      var multipleChoiceData = JSON.parse(
+        JSON.stringify(prevState.multipleChoiceData)
+      );
+      multipleChoiceData[name] = value;
+      return { multipleChoiceData };
+    });
+  };
+
+  handleFreeResponseSectionChange = (bankFormType, event) => {
+    const { name, value } = event.target;
+    if (bankFormType === "checkSlip") {
+      this.setState((prevState) => {
+        var freeResponseData = JSON.parse(
+          JSON.stringify(prevState.freeResponseData)
+        );
+        freeResponseData[bankFormType][name] = value;
+        return { freeResponseData };
+      });
+    } else {
+      this.setState((prevState) => {
+        var freeResponseData = JSON.parse(
+          JSON.stringify(prevState.freeResponseData)
+        );
+        freeResponseData[bankFormType]["rowEntries"][name] = value;
+        return { freeResponseData };
+      });
     }
+  };
+
+  setSchoolData = (schoolData) => {
+    this.setState({ schoolData });
+    // initialise to first school district
+    this.setInitialSchoolInfo(Object.keys(schoolData)[0]);
+  };
+
+  setInitialSchoolInfo = (schoolDistrict) => {
+    var schoolData = this.state.schoolData;
+    var aboutData = JSON.parse(JSON.stringify(this.state.aboutData));
+    var school = schoolData[schoolDistrict][0].id;
+    aboutData = { ...aboutData, schoolDistrict, school };
+    this.setState({ aboutData });
+  };
 
-    handleMultipleSectionChange = event => {
-        const {name, value} = event.target
-        this.setState(prevState => {
-            var multipleChoiceData = JSON.parse(JSON.stringify(prevState.multipleChoiceData));
-            multipleChoiceData[name] = value;
-            return { multipleChoiceData };
-        })
+  handleSubmit = (event) => {
+    event.preventDefault();
+    // about section
+    var aboutSectionAnswerObj = { ...this.state.aboutData };
+    // if other grade is included, submit other value as grade in form
+    if (aboutSectionAnswerObj.otherGrade) {
+      aboutSectionAnswerObj.grade = aboutSectionAnswerObj.otherGrade;
+      aboutSectionAnswerObj.otherGrade = null;
     }
+    // remove school district from response
+    aboutSectionAnswerObj.schoolDistrict = null;
+
+    // format free response data
+    var depositFormData = this.state.freeResponseData.depositForm
+      .rowEntries;
+    var depositFormAnswerObj = {
```

Comment:
something that I was thinking of, these dollar amounts might make more sense to be numbers as opposed to strings. It would make validation and automated grading much easier,

Reply:
I considered this as well. I was thinking it'd make it faster but how would it be easier?